src/TypeSpec.Extension/Emitter.Csharp/src/lib/operation.ts

```diff
@@ -376,55 +379,30 @@ export function loadOperation(
         op: HttpOperation,
         resourceOperation?: ResourceOperation
     ): OperationLongRunning | undefined {
-        if (!isLongRunningOperation(context, op.operation)) return undefined;
-
-        const finalResponse = loadLongRunningFinalResponse(
-            context,
-            op,
-            resourceOperation
-        );
-        if (finalResponse === undefined) return undefined;
+        const metadata = getLroMetadata(program, op.operation);
+        if (metadata === undefined) {
+            return undefined;
+        }
 
         return {
-            FinalStateVia: OperationFinalStateVia.Location, // data plane only supports `location`
-            FinalResponse: finalResponse
+            FinalStateVia: convertLroFinalStateVia(metadata.finalStateVia),
+            FinalResponse: {
+                // in swagger, we allow delete to return some meaningful body content
+                // for now, let assume we don't allow return type
+                StatusCodes: op.verb === "delete" ? [204] : [200],
+                BodyType:
+                    op.verb === "delete"
```

Comment:
what if service side return the response body in the LRO delete call?   Is this possible?

Reply:
@lirenhe It's possible. There is a case in mgmt plane. Let's assume it's not allowed for data plane for now. `getLroMetadata` will change, we'll be clearer how to handle that case in the near future.